spotbugs/src/main/java/edu/umd/cs/findbugs/detect/FindAssertionsWithSideEffects.java

```diff
@@ -0,0 +1,63 @@
+package edu.umd.cs.findbugs.detect;
+
+import org.apache.bcel.Const;
+import edu.umd.cs.findbugs.BugInstance;
+import edu.umd.cs.findbugs.BugReporter;
+import edu.umd.cs.findbugs.ba.SignatureParser;
+import edu.umd.cs.findbugs.util.MutableClasses;
+
+/**
+ * This detector can find assertions that violate the EXP06 rule.
+ */
+public class FindAssertionsWithSideEffects extends AbstractAssertDetector {
+
+    public FindAssertionsWithSideEffects(BugReporter bugReporter) {
+        super(bugReporter);
+    }
+
+    /**
+     * Returns true if the opcode is a method invocation false otherwise
+     */
+    private boolean isMethodCall(int seen) {
+        return seen == Const.INVOKESTATIC ||
+                seen == Const.INVOKEVIRTUAL ||
+                seen == Const.INVOKEINTERFACE ||
+                seen == Const.INVOKESPECIAL ||
+                seen == Const.INVOKEDYNAMIC;
+    }
+
+    /**
+     * Returns true if the opcode is a side effect producing instruction
+     */
+    private boolean checkSeen(int seen) {
+        return seen == Const.IINC ||
+                seen == Const.ISTORE ||
+                seen == Const.ISTORE_0 ||
+                seen == Const.ISTORE_1 ||
+                seen == Const.ISTORE_2 ||
+                seen == Const.ISTORE_3;
+    }
+
+    /**
+     * Finds assertion with possible side effect
+     */
+    @Override
+    protected void detect(int seen) {
+        if (isMethodCall(seen)) {
+            String retSig = new SignatureParser(getXMethodOperand().getSignature()).getReturnTypeSignature();
+            String classSig = getXClassOperand().getSourceSignature();
+            if (MutableClasses.mutableSignature("L" + getClassConstantOperand() + ";") &&
+                    MutableClasses.looksLikeASetter(getNameConstantOperand(), retSig, classSig)) {
```

Comment:
In the method signature the order of the parameters are: methodName, classSig, retSig. In this call you reversed the order of the last two arguments: retSig, then classSig. This makes the code a bit less readable. However these parameters are almost* symmetric - since there is only an equality check in the implementation -, I think it would be nice if the call and the method declaration had the same order of the arguments.

* Only almost symmetric, since the classSig parameter (the middle one) can't be null. Was this your motive for switching the argument order?

Reply:
This can be resolved, since it got fixed.